src/backend/renderer/gles2/mod.rs

```diff
@@ -1627,6 +1632,108 @@ impl Offscreen<Gles2Renderbuffer> for Gles2Renderer {
     }
 }
 
+impl<B1, B2> Blit<B1, B2> for Gles2Renderer
+where
+    Self: Bind<B1> + Bind<B2>,
+{
+    fn blit(
+        &mut self,
+        from: B1,
+        to: B2,
+        src: Rectangle<i32, Physical>,
+        dst: Rectangle<i32, Physical>,
+        filter: TextureFilter,
+    ) -> Result<(), <Self as Renderer>::Error> {
+        // glBlitFramebuffer is sadly only available for GLES 3.0 and higher
+        if self.gl_version < version::GLES_3_0 {
+            return Err(Gles2Error::GLVersionNotSupported(version::GLES_3_0));
+        }
+
+        self.bind(from)?;
+        let src_target = self.target.take().unwrap();
+        self.bind(to)?;
+        let dst_target = self.target.take().unwrap();
+        self.unbind()?;
+
+        match (&src_target, &dst_target) {
+            (&Gles2Target::Surface(ref src), &Gles2Target::Surface(ref dst)) => unsafe {
+                self.egl
+                    .make_current_with_draw_and_read_surface(Some(&**dst), Some(&**src))?;
+            },
+            (&Gles2Target::Surface(ref src), _) => unsafe {
+                self.egl
+                    .make_current_with_draw_and_read_surface(None, Some(&**src))?;
+            },
+            (_, &Gles2Target::Surface(ref dst)) => unsafe {
+                self.egl
+                    .make_current_with_draw_and_read_surface(Some(&**dst), None)?;
+            },
+            (_, _) => unsafe {
+                self.egl.make_current()?;
+            },
+        }
+
+        match &src_target {
+            Gles2Target::Image { ref buf, .. } => unsafe {
+                self.gl.BindFramebuffer(ffi::READ_FRAMEBUFFER, buf.fbo)
+            },
+            Gles2Target::Texture { ref fbo, .. } => unsafe {
+                self.gl.BindFramebuffer(ffi::READ_FRAMEBUFFER, *fbo)
+            },
+            Gles2Target::Renderbuffer { ref fbo, .. } => unsafe {
+                self.gl.BindFramebuffer(ffi::READ_FRAMEBUFFER, *fbo)
+            },
+            _ => {}
+        }
+        match &dst_target {
+            Gles2Target::Image { ref buf, .. } => unsafe {
+                self.gl.BindFramebuffer(ffi::DRAW_FRAMEBUFFER, buf.fbo)
+            },
+            Gles2Target::Texture { ref fbo, .. } => unsafe {
+                self.gl.BindFramebuffer(ffi::DRAW_FRAMEBUFFER, *fbo)
+            },
+            Gles2Target::Renderbuffer { ref fbo, .. } => unsafe {
+                self.gl.BindFramebuffer(ffi::DRAW_FRAMEBUFFER, *fbo)
+            },
+            _ => {}
+        }
+
+        let status = unsafe { self.gl.CheckFramebufferStatus(ffi::FRAMEBUFFER) };
+        if status != ffi::FRAMEBUFFER_COMPLETE {
+            let _ = self.unbind();
+            return Err(Gles2Error::FramebufferBindingError);
+        }
+
+        let errno = unsafe {
+            let _ = self.gl.GetError(); // clear flag before
```

Comment:
Don't we need to drain the error in a loop?

Reply:
https://docs.gl/es2/glGetError

Seems you are right.

Reply:
fixed